From a847fd9b9bc7a6303ff74d6c783cb0b150f79cac Mon Sep 17 00:00:00 2001
From: Krzysztof Struczynski <krzysztof.struczynski@huawei.com>
Date: Fri, 10 Sep 2021 19:59:58 +0800
Subject: [PATCH] ima: Introduce ima-ns-sig template
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: a847fd9b9bc7a6303ff74d6c783cb0b150f79cac


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I49KW1
CVE: NA

--------------------------------

Introduce ima-ns-sig template: "d-ng|n-ng|ns|sig"

Signed-off-by: Krzysztof Struczynski <krzysztof.struczynski@huawei.com>
Reviewed-by: Zhang Tianxing <zhangtianxing3@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 security/integrity/ima/ima_template.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/integrity/ima/ima_template.c b/security/integrity/ima/ima_template.c
index e72497f37bb5..5cd4ccd0d240 100644
--- a/security/integrity/ima/ima_template.c
+++ b/security/integrity/ima/ima_template.c
@@ -23,6 +23,7 @@ static struct ima_template_desc builtin_templates[] = {
 	{.name = "ima-buf", .fmt = "d-ng|n-ng|buf"},
 	{.name = "ima-modsig", .fmt = "d-ng|n-ng|sig|d-modsig|modsig"},
 	{.name = "ima-ns", .fmt = "d-ng|n-ng|ns"},
+	{.name = "ima-ns-sig", .fmt = "d-ng|n-ng|ns|sig"},
 	{.name = "", .fmt = ""},	/* placeholder for a custom format */
 };
 
-- 
2.26.2

